var/spack/repos/builtin/packages/conduit/package.py

```diff
@@ -256,6 +257,80 @@ def check_install(self):
             example = Executable("./conduit_example")
             example()
 
+    # TODO: Replace this method and its 'get' use for cmake path with
+    #   join_path(self.spec['cmake'].prefix.bin, 'cmake') once stand-alone
+    #   tests can access build dependencies through self.spec['cmake'].
+    def cmake_bin(self, set=True):
+        """(Hack) Set/get cmake dependency path."""
+        filepath = join_path(self.install_test_root, "cmake_bin_path.txt")
+        if set:
+            with open(filepath, "w") as out_file:
+                cmake_bin = join_path(self.spec["cmake"].prefix.bin, "cmake")
+                out_file.write("{0}\n".format(cmake_bin))
+        else:
+            with open(filepath, "r") as in_file:
+                return in_file.read().strip()
+
+    @run_after("build")
+    def setup_smoke_test(self):
+        """Setup info needed for spack smoke tests"""
+        # TODO: Remove once self.spec['cmake'] is available in `spack test run`
+        self.cmake_bin(set=True)
+
+    def test(self):
+        """
+        Checks the spack install of conduit using conduit's
+        using-with-cmake and using-with-make examples
+        (supports: spack test run)
+        """
+        # Note: This looks similar to the `self.check_install` logic
+        # but there are subtle differences:
+        # (paths differ , and we use `self.run_test`)
+        print("Checking Conduit installation...")
+        install_prefix = self.spec.prefix
+        test_stage_dir = self.test_suite.test_dir_for_spec(self.spec)
+        #####
+        # using-with-cmake
+        #####
+        example_src_dir = join_path(install_prefix, "examples", "conduit", "using-with-cmake")
+        print("Checking using-with-cmake example...")
+        with working_dir(
+            join_path(test_stage_dir, "check-post-conduit-using-with-cmake-example"), create=True
+        ):
+            opts = ["-DCONDUIT_DIR={0}".format(install_prefix), example_src_dir]
+            opts += self.std_cmake_args
```

Comment:
I think this is going to take a little work on Spack's part I suspect due to multi-build system changes (@alalazo).  I'm getting the following error (erroneously flagged as line 313 but clearly at line 301.

```
$ spack test run conduit
==> Spack test otu22qpxaoovfhbjharde4zis2sjm6mm
==> Testing package conduit-0.8.4-ot273dx
==> Error: AttributeError: 'Conduit' object has no attribute 'std_cmake_args'

The 'conduit' package cannot find an attribute while trying to build from sources. This might be due to a change in Spack's package format to support multiple build-systems for a single package. You can fix this by updating the build recipe, and you can also report the issue as a bug. More information at https://spack.readthedocs.io/en/latest/packaging_guide.html#installation-procedure
...
```

Reply:
@tldahlgren -- @white238 suggested that:

```
opts += self.builder.std_cmake_args
```

may work, due to recent refactor how builders work?

Reply:
Yes, that works.

Reply:
I should say, it lets me install the software and run the tests (which are failing for me) on ruby:

```
==> Testing package conduit-0.8.4-6b5pkng
==> [2022-11-10-15:34:38.157607] Installing /usr/WS1/dahlgren/releases/spack/opt/spack/linux-rhel7-skylake_avx512/gcc-8.3.1/conduit-0.8.4-6b5pkng6a6ia4a3deiqeet3mywyf76a6/.spack/test to /g/g21/dahlgren/.spack/test/usr76buvkbg42uxt7xsvqhaa7ocmocxp/conduit-0.8.4-6b5pkng/cache/conduit
Checking Conduit installation...
Checking using-with-cmake example...
==> [2022-11-10-15:34:38.183300] test: cmake: expect command status in [0]
==> [2022-11-10-15:34:38.183585] '/usr/tce/bin/cmake' '-DCONDUIT_DIR=/usr/WS1/dahlgren/releases/spack/opt/spack/linux-rhel7-skylake_avx512/gcc-8.3.1/conduit-0.8.4-6b5pkng6a6ia4a3deiqeet3mywyf76a6' '/usr/WS1/dahlgren/releases/spack/opt/spack/linux-rhel7-skylake_avx512/gcc-8.3.1/conduit-0.8.4-6b5pkng6a6ia4a3deiqeet3mywyf76a6/examples/conduit/using-with-cmake' '-G' 'Unix Makefiles' '-DCMAKE_INSTALL_PREFIX:STRING=/usr/WS1/dahlgren/releases/spack/opt/spack/linux-rhel7-skylake_avx512/gcc-8.3.1/conduit-0.8.4-6b5pkng6a6ia4a3deiqeet3mywyf76a6' '-DCMAKE_BUILD_TYPE:STRING=RelWithDebInfo' '-DBUILD_TESTING:BOOL=OFF' '-DCMAKE_INTERPROCEDURAL_OPTIMIZATION:BOOL=OFF' '-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON' '-DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=ON' '-DCMAKE_INSTALL_RPATH:STRING=/usr/WS1/dahlgren/releases/spack/opt/spack/linux-rhel7-skylake_avx512/gcc-8.3.1/conduit-0.8.4-6b5pkng6a6ia4a3deiqeet3mywyf76a6/lib;/usr/WS1/dahlgren/releases/spack/opt/spack/linux-rhel7-skylake_avx512/gcc-8.3.1/conduit-0.8.4-6b5pkng6a6ia4a3deiqeet3mywyf76a6/lib64;/usr/WS1/dahlgren/releases/spack/opt/spack/linux-rhel7-skylake_avx512/gcc-8.3.1/hdf5-1.8.22-lccg4en2cfix7qwb5ufmh535xlzd4oym/lib;/usr/WS1/dahlgren/releases/spack/opt/spack/linux-rhel7-skylake_avx512/gcc-8.3.1/openmpi-4.1.4-igcolfhweythdlzfphmoc6cunoommib6/lib;/usr/WS1/dahlgren/releases/spack/opt/spack/linux-rhel7-skylake_avx512/gcc-8.3.1/hwloc-2.8.0-ckovy7uxzlbjzu4b75jprfr2gb7ls5vt/lib;/usr/WS1/dahlgren/releases/spack/opt/spack/linux-rhel7-skylake_avx512/gcc-8.3.1/libpciaccess-0.16-juvuq3pprqubxunmtutl7warvnngo4gg/lib;/usr/WS1/dahlgren/releases/spack/opt/spack/linux-rhel7-skylake_avx512/gcc-8.3.1/libxml2-2.10.1-3hgq3elsdpkp2jotp7vfuynfs5o2awc4/lib;/usr/WS1/dahlgren/releases/spack/opt/spack/linux-rhel7-skylake_avx512/gcc-8.3.1/libiconv-1.16-5b53snqgnucoyydgfpg7qghiute6wvjb/lib;/usr/WS1/dahlgren/releases/spack/opt/spack/linux-rhel7-skylake_avx512/gcc-8.3.1/zlib-1.2.13-fdyvt2xzc5zqng6afhnrmgjski3dwrah/lib;/usr/WS1/dahlgren/releases/spack/opt/spack/linux-rhel7-skylake_avx512/gcc-8.3.1/numactl-2.0.14-s6qjyvdqhiopls2pjrrnkfoja6xc4gm5/lib;/usr/WS1/dahlgren/releases/spack/opt/spack/linux-rhel7-skylake_avx512/gcc-8.3.1/pmix-4.1.2-5hzvt5ishuapqfh275hnhpepjjd4jfqv/lib;/usr/WS1/dahlgren/releases/spack/opt/spack/linux-rhel7-skylake_avx512/gcc-8.3.1/libevent-2.1.12-n5xfdzofptz6g42jm2462pmyroug7ryg/lib;/usr/WS1/dahlgren/releases/spack/opt/spack/linux-rhel7-skylake_avx512/gcc-8.3.1/metis-5.1.0-yfo6nxkd7z3n3jzhvafskt6lb33iuc5u/lib;/usr/WS1/dahlgren/releases/spack/opt/spack/linux-rhel7-skylake_avx512/gcc-8.3.1/parmetis-4.0.3-isva4dypth6t6rejfqdlgry73axrhgjf/lib' '-DCMAKE_PREFIX_PATH:STRING=/usr/WS1/dahlgren/releases/spack/opt/spack/linux-rhel7-skylake_avx512/gcc-8.3.1/parmetis-4.0.3-isva4dypth6t6rejfqdlgry73axrhgjf;/usr/WS1/dahlgren/releases/spack/opt/spack/linux-rhel7-skylake_avx512/gcc-8.3.1/metis-5.1.0-yfo6nxkd7z3n3jzhvafskt6lb33iuc5u;/usr/WS1/dahlgren/releases/spack/opt/spack/linux-rhel7-skylake_avx512/gcc-8.3.1/hdf5-1.8.22-lccg4en2cfix7qwb5ufmh535xlzd4oym;/usr/WS1/dahlgren/releases/spack/opt/spack/linux-rhel7-skylake_avx512/gcc-8.3.1/openmpi-4.1.4-igcolfhweythdlzfphmoc6cunoommib6;/usr/WS1/dahlgren/releases/spack/opt/spack/linux-rhel7-skylake_avx512/gcc-8.3.1/pmix-4.1.2-5hzvt5ishuapqfh275hnhpepjjd4jfqv;/usr/WS1/dahlgren/releases/spack/opt/spack/linux-rhel7-skylake_avx512/gcc-8.3.1/libevent-2.1.12-n5xfdzofptz6g42jm2462pmyroug7ryg;/usr/WS1/dahlgren/releases/spack/opt/spack/linux-rhel7-skylake_avx512/gcc-8.3.1/numactl-2.0.14-s6qjyvdqhiopls2pjrrnkfoja6xc4gm5;/usr/WS1/dahlgren/releases/spack/opt/spack/linux-rhel7-skylake_avx512/gcc-8.3.1/hwloc-2.8.0-ckovy7uxzlbjzu4b75jprfr2gb7ls5vt;/usr/WS1/dahlgren/releases/spack/opt/spack/linux-rhel7-skylake_avx512/gcc-8.3.1/libxml2-2.10.1-3hgq3elsdpkp2jotp7vfuynfs5o2awc4;/usr/WS1/dahlgren/releases/spack/opt/spack/linux-rhel7-skylake_avx512/gcc-8.3.1/zlib-1.2.13-fdyvt2xzc5zqng6afhnrmgjski3dwrah;/usr/WS1/dahlgren/releases/spack/opt/spack/linux-rhel7-skylake_avx512/gcc-8.3.1/libiconv-1.16-5b53snqgnucoyydgfpg7qghiute6wvjb;/usr/WS1/dahlgren/releases/spack/opt/spack/linux-rhel7-skylake_avx512/gcc-8.3.1/libpciaccess-0.16-juvuq3pprqubxunmtutl7warvnngo4gg;/usr/tce'
-- The C compiler identification is GNU 4.9.3
-- The CXX compiler identification is GNU 4.9.3
-- Check for working C compiler: /usr/tce/bin/cc
-- Check for working C compiler: /usr/tce/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/tce/bin/c++
-- Check for working CXX compiler: /usr/tce/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Conduit was built with HDF5 Support
-- Looking for HDF5 at: /usr/WS1/dahlgren/releases/spack/opt/spack/linux-rhel7-skylake_avx512/gcc-8.3.1/hdf5-1.8.22-lccg4en2cfix7qwb5ufmh535xlzd4oym
-- Found HDF5: hdf5-shared (found version "1.8.22")  
-- HDF5_DIR_REAL=/usr/WS1/dahlgren/releases/spack/opt/spack/linux-rhel7-skylake_avx512/gcc-8.3.1/hdf5-1.8.22-lccg4en2cfix7qwb5ufmh535xlzd4oym
-- Checking that found HDF5_INCLUDE_DIRS are in HDF5_DIR
-- HDF5_INCLUDE_DIRS=/usr/WS1/dahlgren/releases/spack/opt/spack/linux-rhel7-skylake_avx512/gcc-8.3.1/hdf5-1.8.22-lccg4en2cfix7qwb5ufmh535xlzd4oym/include
--  /usr/WS1/dahlgren/releases/spack/opt/spack/linux-rhel7-skylake_avx512/gcc-8.3.1/hdf5-1.8.22-lccg4en2cfix7qwb5ufmh535xlzd4oym/include includes HDF5_DIR (/usr/WS1/dahlgren/releases/spack/opt/spack/linux-rhel7-skylake_avx512/gcc-8.3.1/hdf5-1.8.22-lccg4en2cfix7qwb5ufmh535xlzd4oym)
--  /usr/WS1/dahlgren/releases/spack/opt/spack/linux-rhel7-skylake_avx512/gcc-8.3.1/hdf5-1.8.22-lccg4en2cfix7qwb5ufmh535xlzd4oym/include includes HDF5_REAL_DIR ()
-- HDF5 is parallel:  ON
-- Configuring incomplete, errors occurred!
See also "/g/g21/dahlgren/.spack/test/usr76buvkbg42uxt7xsvqhaa7ocmocxp/conduit-0.8.4-6b5pkng/check-post-conduit-using-with-cmake-example/CMakeFiles/CMakeOutput.log".
See also "/g/g21/dahlgren/.spack/test/usr76buvkbg42uxt7xsvqhaa7ocmocxp/conduit-0.8.4-6b5pkng/check-post-conduit-using-with-cmake-example/CMakeFiles/CMakeError.log".
CMake Error at /usr/WS1/dahlgren/releases/spack/opt/spack/linux-rhel7-skylake_avx512/gcc-8.3.1/conduit-0.8.4-6b5pkng6a6ia4a3deiqeet3mywyf76a6/lib/cmake/conduit/conduit_setup_deps.cmake:188 (MESSAGE):
  Cannot find MPI, but the HDF5 Conduit is linked to requires MPI support.
  (HDF5_IS_PARALLEL == TRUE).
Call Stack (most recent call first):
  /usr/WS1/dahlgren/releases/spack/opt/spack/linux-rhel7-skylake_avx512/gcc-8.3.1/conduit-0.8.4-6b5pkng6a6ia4a3deiqeet3mywyf76a6/lib/cmake/conduit/ConduitConfig.cmake:71 (include)
  CMakeLists.txt:59 (find_package)
...
```

Reply:
if hdf5 was built with mpi support (spack default), conduit example will have build with mpi.

two options:

1) we do have "using-zzz-with-mpi" tests install -- however then we need spack stand along test env to locate mpi compilers and know mpi launch commands. 

2) we build hdf5 w/o mpi support (conduit does not use it)


are there recipes we can look at to see if 1) is feasible?

Reply:
The `ParallelNetCDFPackage` uses the compiler and knows how to launch with different commands: https://github.com/spack/spack/blob/0cec2d3110ca76b40f6e7bcb9ae735577298f282/var/spack/repos/builtin/packages/parallel-netcdf/package.py#L180.

You'll want to be sure to tell Spack that you want to use the compilers to build the tests: https://spack.readthedocs.io/en/latest/packaging_guide.html#enabling-test-compilation